src/main/java/com/redhat/pantheon/sling/PantheonRepositoryInitializer.java

```diff
@@ -41,22 +41,29 @@ public void processRepository(SlingRepository slingRepository) throws Exception
     private void initializeRepositoryACLs(SlingRepository slingRepository) throws RepositoryException {
         JackrabbitSession s = (JackrabbitSession) slingRepository.loginAdministrative(null);
         try {
-            User admin = (User) s.getUserManager().getAuthorizable("admin");
-            admin.changePassword("ccsadmin"); // FIXME - hardcoding admin passwords is a Bad Thing
-
+            // Create and give the pantheon service user permissions to the whole /content path
             try {
                 s.getUserManager().createSystemUser("pantheon", null);
+                s.save();
```

Comment:
Do you need this s.save() here, since we've already got an s.save() coming later in the method?

Reply:
Ah, perhaps it's so that you can catch the `AuthorizableExistsException` separately from the rest of the logic?

Reply:
I need the call to save here so the call in line 52 is able to "see" the changes. Otherwise, that call would end up throwing an error because it doesn't recognize the 'pantheon' principal. It acts like a bit of a flush for the operations on the JCR repository.

Reply:
Ah, that also makes sense, thanks!